Exceptions.hpp

```diff
@@ -0,0 +1,9 @@
+#include <stdexcept>
+
+class InvalidGear : public std::runtime_error
+{
+
+    public:
+        InvalidGear() : std::runtime_error("You can not change the current gear to R. Slow down!")
```

Comment:
Be very precise in explanatory string (a.k.a. "what_arg"). Check how this info is formulated for standard exceptions. I suggest something like "Changing gear to "R" is not allowed when velocity != 0".